### PR TITLE
build: add pack & upload CI job

### PR DESCRIPTION
**Pull request by RasPhilCo**
I'd like to wait until the package.json spacing issue is fixed to make sure this PR doesn't erase something needed.

WI forthcoming
#123

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -104,9 +105,9 @@
     "@salesforce/plugin-auth": "1.4.8",
     "@salesforce/plugin-config": "1.2.4",
     "@salesforce/plugin-generator": "^1.1.0",
+    "@salesforce/plugin-telemetry": "1.0.1",
```
Comment (RasPhilCo) [inline package.json:108]: rebase mishap, i'll rm this

File: .circleci/config.yml [hunk 2]
```diff
@@ -51,3 +72,14 @@ workflows:
...
             - approval
+          filters:
+            branches:
+              only:
+                - main
+      - pack-and-upload-tarballs:
+          filters:
+            branches:
+              only:
+                - archives
```
Comment (RasPhilCo) [inline .circleci/config.yml:83]: add `- main`

Comment (amphro) [inline .circleci/config.yml:83]:
```suggestion
                - main
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -51,3 +72,14 @@ workflows:
...
+          requires:
+            - approval
```
Comment (amphro) [inline .circleci/config.yml:85]:
```suggestion
            - release-management/release-package
```

**Review by amphro — APPROVED** [review]

**Pull request merged**